### PR TITLE
Exported function docs & examples

### PR DESCRIPTION
**Pull request by zeebo**
Adds documentation and examples to all the exported functions/types.

### CODE REVIEW

File: template.go [hunk 1]
```diff
@@ -60,6 +68,7 @@ type funcDecl struct {
 	val  reflect.Value
 }
 
+//Template is the type that represents a template.
```
Comment (jbaikge) [inline template.go]: "Template represents a parsed template where all dependencies are attached through other functions."

How's that?

Comment (zeebo) [inline template.go]: But the template isn't parsed until Execute is called.

Comment (jbaikge) [inline template.go]: Perused the Go docs, what you've got follows their style, just wish there were more fireworks to it.

File: example_test.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+func ExampleParse() {
+	t := tmpl.Parse("tmpls/base.tmpl")
+	if err := t.Execute(w, nil); err != nil {
+		panic(err)
+	}
+}
+
+func ExampleTemplate_Blocks() {
+	t := tmpl.Parse("tmpls/base.tmpl")
```
Comment (jbaikge) [inline example_test.go]: I'd stick with "templates" or "tmpl". "tmpls" feels a little abbreviated to me.

File: template.go (hunk 1 above)
Comment (zeebo) [inline template.go]: I like adding a note about the dependencies being attached later.

File: example_test.go (hunk 2 above)
Comment (zeebo) [inline example_test.go]: Roger.

**Comment by zeebo** [comment]
PTAL.

**Comment by jbaikge** [comment]
Re #21: This is ready to go, will it close the issue?

**Comment by zeebo** [comment]
It will if we name the merge commit right. Is this ready to go?

**Comment by zeebo** [comment]
Haha maybe I should read better!

**Pull request merged**